hoomd/pytest/test_simulation.py

```diff
@@ -0,0 +1,172 @@
+import hoomd
+import numpy as np
+import pytest
+from copy import deepcopy
+try:
+    import gsd.hoomd
+    skip_gsd = False
+except ImportError:
+    skip_gsd = True
+
+
+@pytest.fixture(scope="function")
+def get_snapshot(device):
+    def make_snapshot(n=10, particle_types=['A']):
+        s = hoomd.snapshot.Snapshot(device.comm)
+        if s.exists:
+            s.configuration.box = [20, 20, 20, 0, 0, 0]
+            s.particles.N = n
+            s.particles.position[:] = np.random.uniform(-10, 10, size=(n, 3))
+            s.particles.types = particle_types
+        return s
+    return make_snapshot
+
+
+def make_gsd_snapshot(hoomd_snapshot):
```

Comment:
Instead of creating the list of top level properties, why don't you do something like,

````python
for att in all_attr:
    if cond:
        for prop in dir(getattr(hoomd_snapshot, att)):
            if cond:
                setattr()
````

Reply:
Fixed 68371cc68